src/analyses/malloc_null.ml

```diff
@@ -94,7 +94,7 @@ struct
   (* Remove null values from state that are unreachable from exp.*)
   let remove_unreachable (ask: Queries.ask) (args: exp list) (st: D.t) : D.t =
     let reachable =
-      let do_exp e a =
+      let do_exp a e =
```

Comment:
This seems very odd, the return value depends on the order of the elements in `args`: As soon as a top pointer is encountered, AD is reset to `{}` but accumulation continues as normal after that?

Reply:
Intuitively, one would like to:

- return `a` instead of `AD.empty ()`
- return some special value indicating that everything is to be considered reachable.